### PR TITLE
Add Chisel Eslint and Prettier configuration

### PR DESCRIPTION
**Pull request by arturkot**
Initial attempt to add our [own Chisel config](https://github.com/xfiveco/eslint-config-chisel).

### CODE REVIEW

**Comment by arturkot** [comment]
Hi @luboskmetko @jakub300 ,

Questions/concerns:

* I guess we should publish the config on NPM first, before merging this PR?
* Should I add more reviewers?

**Comment by jakub300** [comment]
Maybe we should set `trailingComma` to es6 and enforce us of babel (remove project creation option)?

File: generators/app/templates/package.json [hunk 1]
```diff
@@ -78,7 +78,8 @@
     "through2": "^2.0.3",
     "vinyl-buffer": "^1.0.0",
     "vinyl-source-stream": "^1.1.0",
-    "watchify": "^3.7.0"
+    "watchify": "^3.7.0",
+    "eslint-config-chisel": "https://github.com/xfiveco/eslint-config-chisel"
```
Comment (jakub300) [inline generators/app/templates/package.json]: Dependencies are ordered alphabetically

**Comment by arturkot** [comment]
@jakub300 You mean setting `trailingComma` to `all` I guess? I've got nothing against it.

What I'm a bit afraid of is that with those new Eslint rules we make Chisel compatible with the newer JS versions by default. It's not a bad thing _for me_ but I realise there are people still writing in ES5 and Eslint wuld throw errors left and right in their case (replace var with let, etc.).

On the other hand perhaps it's time to officially switch to ES6 (and newer) versions? Enabling Babel by default is ok IMO. Eslint would only help people transitioning by giving hints of what should be fixed (at least on some basic level...).

cc @marcinkrzeminski  @luboskmetko Ping! Just wanted to get your attention, quite important stuff being discussed. :)

**Comment by luboskmetko** [comment]
I'm ok if we use this as part of transition to ES6 and make Eslint give some hints to developers. One question is of those would be errors preventing build or just warnings?

Another thing is what to do with the current option when setting up the project where you can still switch off Babel/ES6 support

**Comment by arturkot** [comment]
@luboskmetko @jakub300 

Hi guys,

just wanted to let you know I replaced GH link for a real  `eslint-config-chisel` dependency. As for the generator setup I suggest to:

* install `eslint-config-chisel` only if Babel has been selected in options,
* always install Prettier so it could be used as an optional tool.

**Comment by arturkot** [comment]
@luboskmetko @jakub300 

**Update:**

* esling-config-chisel is installed only if Babel has been selected first
* I added auto-formatting of example JS files with Prettier to avoid errors from our own Eslint config. ;) Plus it takes care of silly problems caused by EJS templates.

**Comment by jakub300** [comment]
Dependencies are alphabetically ordered.

I don't like that auto formatting during creating, could we just update source files?

**Comment by arturkot** [comment]
> I don't like that auto formatting during creating, could we just update source files?

Od course we can.

Just thought this auto formatting is more bulletproof. Less hassle with adjusting "proper" EJS tags plus significantly smaller chance of the example files being highlighted by Eslint (which would be kind of ironic ;-> ). It's really not that important as those files are deleted/overwritten in 99% of cases I guess.

What is it that you dislike the most? I'm guessing that it adds additional dependency and increases overall generator complexity?

**Comment by jakub300** [comment]
Can we also set fixed version of prettier (per [docs](https://github.com/prettier/prettier)) and eslint with `~` (per [docs](https://github.com/eslint/eslint))?

**Comment by jakub300** [comment]
Also please update tests

**Comment by jakub300** [comment]
Should prettier be used when ES6 is not selected?

**Comment by arturkot** [comment]
> Dependencies are alphabetically ordered.

Sorted!

> I don't like that auto formatting during creating, could we just update source files?

Done. 👍 

> Can we also set fixed version of prettier (per docs) and eslint with ~ (per docs)?

`package.json` updated.

> Also please update tests

You betcha!

> Should prettier be used when ES6 is not selected?

I think it should! I decided to install it by default and add the (Prettier) config. You _don't have to use it_ but it works nice with ES5 and is there to help you out with formatting. Even if someone starts a project without Babel and doesn't mind Prettier, someone else can come in later and trigger it to make files more readable. Not that uncommon scenario. ;)

**Comment by jakub300** [comment]
Hey @arturkot, could you update the documentation?
At minimum, mentioning Prettier in the Features and updating Project structure. I think we should also write few words about prettier but we may do it later.

**Comment by jakub300** [comment]
Aaaand can we please remove first empty line and left one empty line at the end of .eslintrc

**Comment by jakub300** [comment]
Please ignore `/gulp` directory and `/gulpfile.js`.

**Comment by jakub300** [comment]
Hey @arturkot, please also update prettier to newest version.

Hey @luboskmetko, I'd like this to be merged this week, could you find a moment and take a look?

**Comment by luboskmetko** [comment]
@jakub300 feel free to merge it, I trust your judgment on whatever you are doing here, guys. I haven't used Prettier yet, so I'll have to get familiar with it first but leaving this for the time when updates on website are ready. Thanks

**Comment by arturkot** [comment]
Hi @jakub300 @luboskmetko ,

thanks for you comments! Will take care of your updates tomorrow most likely.

**Comment by arturkot** [comment]
Hey @jakub300 ,

all done! There's one extra thing I've decided to add: Eslint and Prettier config for `gulp/` directory. I think those Eslint warnings are quite useful, certainly better than completely disabled linter.

**Pull request merged**